contrib/libbsd/patches/no_nlist.patch

```diff
@@ -0,0 +1,11 @@
+--- configure.ac	2024-01-08 02:16:08.000000000 +0100
++++ configure.ac	2024-02-03 00:40:40.012618732 +0100
+@@ -259,7 +259,7 @@
+ need_bsd_getopt=yes
+ need_progname=yes
+ need_md5=yes
+-need_nlist=yes
++need_nlist=no
```

Comment:
i don't see anyone else needing this patch so what is it for?

Reply:
we don't have <nlist.h> and according to q66 there are no plans to add it to musl-bsd-headers

Reply:
there is probably a cleaner way to do it from the autoreconf cli interface

Reply:
try ac_cv_need_nlist=no

Reply:
from what i see though that doesn't matter? in alpine for instance nlist.h only comes from libbsd itself too, and nothing has to be passed

Reply:
yeah, this isn't needed at all for anything and builds fine

the test fails (

```
FAIL: nlist
===========

Assertion failed: rc == 0 (../../test/nlist.c: main: 72)
FAIL nlist (exit status: 134)
```
)

but that is because of lto, it works with `!lto`. this patch has nothing to do with it

Reply:
also something segfaults silently in install

```
--- install-exec-hook ---
if [ "/usr/lib" != "/usr/lib" ]; then  /builddir/libbsd-0.11.8/build-aux/install-sh -c -d /destdir/libbsd-0.11.8/usr/lib;  mv /destdir/libbsd-0.11.8/usr/lib/libbsd*.so.*  /destdir/libbsd-0.11.8/usr/lib/;  fi
soname=`readlink /destdir/libbsd-0.11.8/usr/lib/libbsd.so`;   /destdir/libbsd-0.11.8/usr/lib/libbsd.so;  (echo '/* GNU ld script';  echo ' * The MD5 functions are provided by the libmd library. */';  cat format.ld;  echo "GROUP(/usr/lib/$soname AS_NEEDED(-lmd))";  )>/destdir/libbsd-0.11.8/usr/lib/libbsd.so
Segmentation fault (core dumped)
```

Reply:
looking real quick the segfault is in libbsd itself somewhere, you'll have to figure out what that is